src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String BUY_AMOUNT = "buy";
+    private static final String SUPPLY_AMOUNT = "supply";
+    private static final String COMMA_SEPARATOR = ",";
+    private static final String RESULT = "result";
+
+    public static void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFromFile(fromFileName);
+        String report = getReport(dataFromFile);
+        writeToFile(toFileName, report);
+    }
+
+    private static String getReport(String fromFileName) {
+        String[] words = fromFileName.split(COMMA_SEPARATOR);
+        StringBuilder stringBuilder = new StringBuilder();
+        int resultSupply = 0;
+        int resultBuy = 0;
+        for (int i = 0; i < words.length; i++) {
+            if (words[i].equals(SUPPLY_AMOUNT)) {
+                resultSupply += Integer.parseInt(words[i + 1]);
+            }
+            if (words[i].equals(BUY_AMOUNT)) {
+                resultBuy += Integer.parseInt(words[i + 1]);
+            }
+        }
+        int res = resultSupply - resultBuy;
+        StringBuilder builder = new StringBuilder();
+        String report = builder.append(SUPPLY_AMOUNT).append(COMMA_SEPARATOR)
+                .append(String.valueOf(resultSupply)).append(System.lineSeparator())
+                .append(BUY_AMOUNT).append(COMMA_SEPARATOR).append(String.valueOf(resultBuy))
+                .append(System.lineSeparator())
+                .append(RESULT).append(COMMA_SEPARATOR).append(String.valueOf(res)).toString();
+        return report;
```

Comment:
```suggestion
        return SUPPLY_AMOUNT + COMMA_SEPARATOR +
                resultSupply + System.lineSeparator() +
                BUY_AMOUNT + COMMA_SEPARATOR + resultBuy +
                System.lineSeparator() +
                RESULT + COMMA_SEPARATOR + res;
```

Reply:
or with StringBuilder:
```suggestion
        return new StringBuilder().append(SUPPLY_AMOUNT).append(COMMA_SEPARATOR)
                .append(resultSupply).append(System.lineSeparator())
                .append(BUY_AMOUNT).append(COMMA_SEPARATOR).append(resultBuy)
                .append(System.lineSeparator())
                .append(RESULT).append(COMMA_SEPARATOR).append(res).toString();
```

But anyway no need of additional variables and String.valueOf()